# Solution to Assignment 13

## 1. Write a Makefile that aligns the reads to the genome and creates BAM and BigWig files.

## 2. Run a feature counter to create a count matrix for your data. 
- The final result of your code should be a count matrix that summarizes read counts for each dataset.
- Include IGV screenshots that demonstrate your data is RNA-Seq data.

## 3. Discuss a few lines of the resulting count matrix. 
- Visually identify rows where the counts show consistent gene expression levels. 
- Try to visually verify that the counts in the count matrix are consistent with the numbers you can observe in the alignment tracks.