#
# Short read alignments.
#
# Docs: https://www.biostarhandbook.com/
#

# Primary variables.

# Genome accession from GenBank.
ACC = AF086833

# SRR number.
SRR = SRR1972918

# The genome label.
LABEL = ebola_mayinga_1976

# The sample name.
SAMPLE = G5731.1

# Additional variables set below.

# The reference genome.
REF = refs/${LABEL}.fa

# The annotation file.
GFF = refs/${LABEL}.gff

# The read pairs.
R1 = reads/${SAMPLE}_1.fq
R2 = reads/${SAMPLE}_2.fq

# The alignment file.
BAM = bam/${SAMPLE}.bam

# The variants with bcftools.
VCF = vcf/${SAMPLE}.vcf.gz

# Number of CPUS
NCPU = 4

# Makefile customizations.
SHELL := bash
.DELETE_ON_ERROR:
.ONESHELL:
.SHELLFLAGS := -eu -o pipefail -c
MAKEFLAGS += --warn-undefined-variables --no-print-directory

usage:
	@echo "#"
	@echo "# Short read alignments."
	@echo "#"
	@echo "# Primary variables."
	@echo "#"
	@echo "# ACC=${ACC}"
	@echo "# SRR=${SRR}"
	@echo "# LABEL=${LABEL}"
	@echo "# SAMPLE=${SAMPLE}"
	@echo "#"
	@echo "# Values generated from primary variables."
	@echo "#"
	@echo "# REF=${REF}"
	@echo "# GFF=${GFF}"
	@echo "# R1=${R1}"
	@echo "# R2=${R2}"
	@echo "# BAM=${BAM}"
	@echo "#"
	@echo "# make run"
	@echo "#"

run:
	# Download the reference genome.
	make -f src/run/genbank.mk \
		ACC=${ACC} \
		REF=${REF} \
		GFF=${GFF} \
		fasta gff

	# Download the short reads.
	make -f src/run/sra.mk \
		SRR=${SRR} \
		N=2500 \
		R1=${R1} \
		R2=${R2} \
		run

	# Index the genome then align the reads to the reference genome.
	make -f src/run/bwa.mk \
		R1=${R1} \
		R2=${R2} \
		REF=${REF} \
		BAM=${BAM} \
		NCPU=${NCPU} \
		index run

# Remove the variant calls generated by the pipeline
run!:
	rm -f ${VCF}

clean: run!

.PHONY: usage run run! clean